sdks/python/apache_beam/coders/coders.py

```diff
@@ -280,10 +280,18 @@ def from_runner_api(cls, coder_proto, context):
     Prefer registering a urn with its parameter type and constructor.
     """
     parameter_type, constructor = cls._known_urns[coder_proto.spec.spec.urn]
-    return constructor(
-        proto_utils.parse_Bytes(coder_proto.spec.spec.payload, parameter_type),
-        [context.coders.get_by_id(c) for c in coder_proto.component_coder_ids],
-        context)
+    try:
+      return constructor(
+          proto_utils.parse_Bytes(
+              coder_proto.spec.spec.payload, parameter_type),
+          [context.coders.get_by_id(c)
+           for c in coder_proto.component_coder_ids],
+          context)
+    except:
+      from apache_beam.transforms.core import RunnerAPICoderHolder
+      if RunnerAPICoderHolder.is_external_coder(coder_proto):
+        return RunnerAPICoderHolder(coder_proto)
```

Comment:
I would prefer context to have a flag whether we allow these holders for unknown transforms. Typically we'd want it to fail early and loud when we encounter something we don't understand.